.ci/install_kata_image.sh

```diff
@@ -6,56 +6,128 @@
 export ROOTFS_DIR
 
+packaged_image="kata-containers-image"
+image_path="/usr/share/kata-containers"
+image_name="kata-containers.img"
+initrd_image_name="kata-containers-initrd.img"
+
+agent_path="${GOPATH}/src/github.com/kata-containers/agent"
+
+# Build Kata agent
+bash -f ${cidir}/install_agent.sh
+agent_commit=$(git --work-tree=${agent_path}/ --git-dir=${agent_path}/.git log --format=%h -1 HEAD)
+
 finish() {
-  [ -d "${ROOTFS_DIR}" ] && [[ "${ROOTFS_DIR}" = *"rootfs"* ]] && sudo rm -rf "${ROOTFS_DIR}"
-  rm -rf "$tmp_dir"
+	[ -d "${ROOTFS_DIR}" ] && [[ "${ROOTFS_DIR}" = *"rootfs"* ]] && sudo rm -rf "${ROOTFS_DIR}"
+	rm -rf "$tmp_dir"
 }
 
 trap finish EXIT
 
-OSBUILDER_DISTRO=${OSBUILDER_DISTRO:-clearlinux}
-AGENT_INIT=${AGENT_INIT:-no}
-TEST_INITRD=${TEST_INITRD:-no}
+get_packaged_agent_version() {
+	version=$(ls "$image_path" | grep "$packaged_image" | cut -d'_' -f4 | cut -d'.' -f1)
+	if [ -z "$version" ]; then
+		die "unknown agent version"
+	fi
+	echo "$version"
+}
 
-# Build Kata agent
-bash -f ${cidir}/install_agent.sh
-agent_commit=$(git --work-tree=$GOPATH/src/github.com/kata-containers/agent/ --git-dir=$GOPATH/src/github.com/kata-containers/agent/.git log --format=%h -1 HEAD)
+install_packaged_image() {
+	if [ "$ID"  == "ubuntu" ]; then
+		sudo -E apt install -y "$packaged_image"
+	elif [ "$ID"  == "fedora" ]; then
+		sudo -E dnf install -y "$packaged_image"
+	elif [ "$ID"  == "centos" ]; then
+		sudo -E yum install -y "$packaged_image"
+	else
+		die "Linux distribution not supported"
+	fi
+}
+
+update_agent() {
+	pushd "$agent_path"
+
+	packaged_image="${image_path}/${image_name}"
+	mount_dir="$(pwd)/mount_dir"
+	mkdir "$mount_dir"
 
-osbuilder_repo="github.com/kata-containers/osbuilder"
+	loop=$(sudo losetup -f --show $packaged_image)
+	sudo partprobe "$loop"
+	sudo mount "${loop}p1" "$mount_dir"
 
-# Clone os-builder repository
-go get -d ${osbuilder_repo} || true
+	echo "Old agent version:"
+	"${mount_dir}/usr/bin/kata-agent" --version
 
-pushd "${GOPATH}/src/${osbuilder_repo}/rootfs-builder"
-sudo -E AGENT_INIT=${AGENT_INIT} AGENT_VERSION=${agent_commit} GOPATH=$GOPATH USE_DOCKER=true ./rootfs.sh ${OSBUILDER_DISTRO}
-popd
+	echo "Install new agent"
+	sudo -E PATH="$PATH" bash -c "make install DESTDIR=$mount_dir"
 
-# Build the image
-if [ x"${TEST_INITRD}" == x"yes" ]; then
-    pushd "${GOPATH}/src/${osbuilder_repo}/initrd-builder"
-    sudo -E AGENT_INIT=${AGENT_INIT} USE_DOCKER=true ./initrd_builder.sh "$ROOTFS_DIR"
-    image_name="kata-containers-initrd.img"
-else
-    pushd "${GOPATH}/src/${osbuilder_repo}/image-builder"
-    sudo -E AGENT_INIT=${AGENT_INIT} USE_DOCKER=true ./image_builder.sh "$ROOTFS_DIR"
-    image_name="kata-containers.img"
-fi
+	echo "New agent version:"
+	"${mount_dir}/usr/bin/kata-agent" --version
 
-# Install the image
-commit=$(git log --format=%h -1 HEAD)
-date=$(date +%Y-%m-%d-%T.%N%z)
-image="kata-containers-${date}-osbuilder-${commit}-agent-${agent_commit}"
+	sudo umount "$mount_dir"
```

Comment:
what about this to a function and also call it if the script exits on error, to avoid let mounted the image?